src/test/HostActivationTests/GivenThatICareAboutHostVersionCompatibility.cs

```diff
@@ -0,0 +1,179 @@
+        public void Old_Host_Is_Forward_Compatible_With_Latest_Runtime(TestProjectFixture previousVersionFixture)
+        {
+            TestProjectFixture fixture = FixtureLatest.Copy();
+            string appExe = fixture.TestProject.AppExe;
+
+            Assert.NotEqual(fixture.Framework, previousVersionFixture.Framework);
+            Assert.NotEqual(fixture.RepoDirProvider.MicrosoftNETCoreAppVersion, previousVersionFixture.RepoDirProvider.MicrosoftNETCoreAppVersion);
+
+            // Baseline (no changes)
+            Command.Create(appExe)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World")
+                .And
+                .HaveStdErrContaining($"--- Invoked apphost [version: {fixture.RepoDirProvider.MicrosoftNETCoreAppVersion}");
+
+            // Use the older apphost and hostfxr
+            // This emulates the case when:
+            //  1) One-off deployment of older runtime (not in global location)
+            //  2) Older apphost executed, but found newer runtime because of multi-level lookup
+            //     Note that we currently don't have multi-level on hostfxr so we will always find the older\one-off hostfxr
+            File.Copy(previousVersionFixture.TestProject.AppExe, fixture.TestProject.AppExe, true);
+            File.Copy(previousVersionFixture.TestProject.HostFxrDll, fixture.TestProject.HostFxrDll, true);
+            Command.Create(appExe)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World")
+                .And
+                .HaveStdErrContaining($"--- Invoked apphost [version: {previousVersionFixture.RepoDirProvider.MicrosoftNETCoreAppVersion}");
+        }
+
+        private static TestProjectFixture CreateTestFixture(string testName, string netCoreAppFramework, string mnaVersion)
+        {
+            var repoDirectories = new RepoDirectoriesProvider(microsoftNETCoreAppVersion: mnaVersion);
+
+            // Use standalone instead of framework-dependent for ease of deployment.
+            var publishFixture = new TestProjectFixture(testName, repoDirectories, framework: netCoreAppFramework, assemblyName: "StandaloneApp");
+
+            try 
+            {
+                publishFixture
+                    .EnsureRestoredForRid(publishFixture.CurrentRid, repoDirectories.CorehostPackages)
+                    .PublishProject(runtime: publishFixture.CurrentRid);
+            }
+            catch (BuildFailureException)
+            {
+                // Some RIDs didn't exist in 2.0+ so if we get an error publishing, ignore.
```

Comment:
Would it be more appropriate to block on the list of RIDs up front? That way we know exactly which RIDs don't work, and for the RIDs we do expect to work, they always will work.

Reply:
I did consider that, but since the list of RIDs can change over time (in master branch), I want to avoid having to change the code whenever a new RID comes out that is not present in 2.0\2.1

Reply:
You could white-list it - here are the list of RIDs that should work.  If the current RID isn't in that list, then skip it.

Reply:
I whitelisted Ubuntu 16. Don't really see the need to test all variants anyway.